src/main/java/org/jenkinsci/plugins/workflow/job/properties/PipelineTriggersJobProperty.java

```diff
@@ -145,14 +149,30 @@ public PipelineTriggersJobProperty reconfigure(@Nonnull StaplerRequest req, @Che
             throw new Descriptor.FormException(e, null);
         }
 
-        this.stopTriggers();
+        PipelineTriggersJobProperty oldProp = owner.getTriggersJobProperty();
+
+        try {
+            owner.removeProperty(this);
+            PipelineTriggersJobProperty thisProp = new PipelineTriggersJobProperty(new ArrayList<Trigger>(trigList.toList()));
+
+            owner.addTriggersJobPropertyWithoutStart(thisProp);
 
-        PipelineTriggersJobProperty thisProp = new PipelineTriggersJobProperty(new ArrayList<Trigger>(trigList.toList()));
-        thisProp.setOwner(owner);
+            thisProp.startTriggers(true);
+            return thisProp;
+        } catch (IOException e) {
+            LOGGER.log(Level.WARNING, "could not configure triggers", e);
+        }
 
-        thisProp.startTriggers(true);
+        if (owner.getTriggersJobProperty() == null && oldProp != null) {
+            try {
+                owner.addTriggersJobPropertyWithoutStart(oldProp);
+                oldProp.startTriggers(true);
+            } catch (IOException e) {
+                LOGGER.log(Level.WARNING, "could not revert to original configured triggers", e);
```

Comment:
Yeah, I'm wary of throwing a `FormException` - I'd rather just revert and warn, but I'm not locked into that.